src/include/rbd/librbd.hpp

```diff
@@ -102,6 +107,27 @@ class CEPH_RBD_API RBD
   const RBD& operator=(const RBD& rhs);
 };
 
+class CEPH_RBD_API ImageOptions {
+public:
+  ImageOptions();
+  ImageOptions(rbd_image_options_t opts);
+  ~ImageOptions();
+
+  int set(int optname, const std::string& optval);
+  int set(int optname, uint64_t optval);
+  int get(int optname, std::string& optval);
+  int get(int optname, uint64_t& optval);
```

Comment:
could we make the out parameter explicit by making it a pointer instead of a reference? I realize we don't always follow this convention (esp. internally), but for a public api we should try to be consistent.